src/SourceBuild/content/repo-projects/razor.proj

```diff
@@ -4,6 +4,8 @@
     <!-- The toolset compiler doesn't get killed with 'build-server shutdown'.
          Instead of disabling shared compilation, disable the toolset compiler package. -->
     <BuildArgs>$(BuildArgs) /p:UsingToolMicrosoftNetCompilers=false</BuildArgs>
+    <!-- Pass down the target RID so we can create the correct apphost for the language servers. -->
+    <BuildArgs>$(BuildArgs) /p:TargetRid=$(TargetRid)</BuildArgs>
```

Comment:
Based on what's described in https://github.com/dotnet/razor/pull/10017#issuecomment-1977517631, can this be passed from the outer build in the razor repo itself (rather than from the VMR)? Would that approach eliminate the need for this PR?

Reply:
I think it's the other way around. These lines in the razor PR shouldn't be necessary after this change is merged: https://github.com/dotnet/razor/pull/10017/files#diff-6a28d92de8923be8821f2123541dc23c8ad0359c96117709e6d49e752c0303d2R21-R23